### DISCUSSION TITLE
Warning: To load an ES module, set "type": "module" in the package.json or use the .mjs extension.

### DISCUSSION THREAD

**Discussion by donglingxi** [statelyai/xstate#3394] [Q&A]
Hi, I tried to use xstate in NuxT3, but after introducing it, the console reported the following error:

<img width="931" alt="image" src="https://user-images.githubusercontent.com/15939094/172829768-37782e2f-d797-4183-9019-56761e7f021b.png">

I tried adding type to package.json, but it still didn't work

**Comment by Andarist** [comment]
Please always try to share a repro case in a runnable form - either by providing a git repository to clone or a codesandbox. OSS maintainers usually can't afford the time to set up the repro, even if exact steps are given.

**Reply by donglingxi** [reply]
It's already created

https://github.com/donglingxi/nuxt3-xstate

I simply initialized it and referenced xstate in the app file, and then got an error after referencing it

**Reply by donglingxi** [reply]
After some exploration, I learned that xState will have some problems supporting ESM at present, and I also saw the previous reply, it seems that XState will not consider supporting ESM for the time being. Since I am using Vite, the new question becomes how can Vite better support XState, I will solve the following problems by myself.

Thank you very much.
